base/shell/explorer/traywnd.cpp

```diff
@@ -2946,59 +2717,118 @@ class CTrayWindow :
-           get pressed when the user clicked left or below the button */
-
-        POINT pt = {GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam)};
-        WINDOWINFO wi = {sizeof(WINDOWINFO)};
-
-        bHandled = FALSE;
-
-        RECT rcStartBtn;
-        m_StartButton.GetWindowRect(&rcStartBtn);
-
-        GetWindowInfo(m_hWnd, &wi);
+        if (!ppt)
+            return FALSE;
+        if (!prcStartBtn)
+            return FALSE;
+        if (!pwi)
+            return FALSE;
 
         switch (m_Position)
         {
             case ABE_TOP:
             case ABE_LEFT:
             {
-                if (pt.x > rcStartBtn.right || pt.y > rcStartBtn.bottom)
-                    return 0;
+                if (ppt->x > prcStartBtn->right || ppt->y > prcStartBtn->bottom)
+                    return FALSE;
                 break;
             }
             case ABE_RIGHT:
             {
-                if (pt.x < rcStartBtn.left || pt.y > rcStartBtn.bottom)
-                    return 0;
+                if (ppt->x < prcStartBtn->left || ppt->y > prcStartBtn->bottom)
+                    return FALSE;
 
-                if (rcStartBtn.right + (int)wi.cxWindowBorders * 2 + 1 < wi.rcWindow.right &&
-                    pt.x > rcStartBtn.right)
+                if (prcStartBtn->right + (int)pwi->cxWindowBorders * 2 + 1 < pwi->rcWindow.right &&
+                    ppt->x > prcStartBtn->right)
                 {
-                    return 0;
+                    return FALSE;
                 }
                 break;
             }
             case ABE_BOTTOM:
             {
-                if (pt.x > rcStartBtn.right || pt.y < rcStartBtn.top)
-                    return 0;
+                if (ppt->x > prcStartBtn->right || ppt->y < prcStartBtn->top)
+                    return FALSE;
 
-                if (rcStartBtn.bottom + (int)wi.cyWindowBorders * 2 + 1 < wi.rcWindow.bottom &&
-                    pt.y > rcStartBtn.bottom)
+                if (prcStartBtn->bottom + (int)pwi->cyWindowBorders * 2 + 1 < pwi->rcWindow.bottom &&
+                    ppt->y > prcStartBtn->bottom)
                 {
-                    return 0;
+                    return FALSE;
                 }
 
                 break;
             }
         }
+        return TRUE;
+    }
+
+    BOOL IsPointWithinShowDesktopButton(LPPOINT ppt, LPRECT prcShowDesktopBtn, WINDOWINFO* pwi)
+    {
+        if (!ppt)
+            return FALSE;
+        if (!prcShowDesktopBtn)
+            return FALSE;
+        UNREFERENCED_PARAMETER(pwi);
+        /*if (!pwi)
+            return FALSE;*/
+        //InflateRect(prcShowDesktopBtn, (int)pwi->cxWindowBorders * 2 + 1, (int)pwi->cyWindowBorders * 2 + 1);
+
+        switch (m_Position)
+        {
+            case ABE_LEFT:
+            {
+                if (ppt->x > prcShowDesktopBtn->right || ppt->y < prcShowDesktopBtn->top)
+                    return FALSE;
```

Comment:
If nothing else is needed in these cases or after the switch, I might suggest this:
```c
return ((ppt->x <= prcShowDesktopBtn->right) && (ppt->y >= prcShowDesktopBtn->top));
```
and similar for the other cases below, instead of the "if (test) return FALSE; break; .... then return TRUE;"
(NOTE that for this, the tests have to be negated...)